### ISSUE TITLE
Version 0.4.0 makes JS code that breaks at load time on Safari 14

### ISSUE THREAD

**Issue by gthb** [cloudflare/serde-wasm-bindgen#28]
We just had to swiftly revert our dependency on `serde-wasm-bindgen` back to 0.3.1, as [the introduction of BigInt support](https://github.com/cloudflare/serde-wasm-bindgen/pull/24) makes `BigUint64Array` and `BigInt64Array` be referenced at module scope in the `*_bg.js` wrapper, causing the module to fail to load in Safari 14 and earlier. [Safari does not have these types until version 15.0](https://caniuse.com/mdn-javascript_builtins_biguint64array) (released last Sep 20th, so there's still lots of Safari 14 out there).

This can break the whole web application (depending on one's bundling, I guess), even one that makes no use of this new BigInt support,  in those browser versions.

So version 0.4 (and the README, if this stays as-is) should really carry a nice big warning about this effect on browser support.

**Comment by RReverser** [comment]
cc @mfish33

This is an interesting one, and a bit unfortunate. I'm surprised those are referenced at top-level scope, I guess we need some way to retrieve them lazily instead.

**Comment by mfish33** [comment]
@RReverser I think this is upstream from us. In the generated code by wasm bindgen these three lines are the culprit.

```js
const u32CvtShim = new Uint32Array(2);
const int64CvtShim = new BigInt64Array(u32CvtShim.buffer);
const uint64CvtShim = new BigUint64Array(u32CvtShim.buffer);
```

Defined in:
https://github.com/rustwasm/wasm-bindgen/blob/ac87c8215bdd28d6aa0e12705996238a78227f8c/crates/cli-support/src/js/mod.rs#L1905
and 
https://github.com/rustwasm/wasm-bindgen/blob/ac87c8215bdd28d6aa0e12705996238a78227f8c/crates/cli-support/src/js/binding.rs#L685

Further, it seems that this issue does not exactly have to do with Bigint types rather having rust bindings that receive u64/i64's as arguments or JS bindings that return u64/i64's

**Comment by RReverser** [comment]
> Further, it seems that this issue does not exactly have to do with Bigint types rather having rust bindings that receive u64/i64's as arguments or JS bindings that return u64/i64's

Right I see, and we introduced those for our `BigInt` conversion bindings... I guess we need to either report this to wasm-bindgen and ask for those variables to be defined only lazily when u64/i64 is _actually_ used, or we need to add a static feature in this crate as a way to opt-out.

**Comment by Liamolucko** [comment]
This should be fixed as of rustwasm/wasm-bindgen#3037.

**Comment by RReverser** [comment]
Great!
